Source/Charts/Components/MarkerView.swift

```diff
@@ -36,18 +36,18 @@ open class MarkerView: NSUIView, IMarker
         {
             offset.x = -point.x
         }
-        else if chart != nil && point.x + width + offset.x > chart!.bounds.size.width
+        else if let chart = chart, point.x + width + offset.x > chart.bounds.size.width
```

Comment:
`guard let chart = chartView else { return }` at the beginning.

Reply:
this func must return a `CGPoint`

Reply:
`guard let chart = chartView else { return offset }`

Reply:
done